### PR TITLE
Add SEH/probes to user-accessible kdbg functions

### PR DESCRIPTION
**Pull request by ThFabba**
## Purpose

Fix bugcheck when running ntdll_winetest:exception.

JIRA issue: [CORE-14057](https://jira.reactos.org/browse/CORE-14057)

This is in "working" state but will need some cleanup (consistency, comments etc).

### CODE REVIEW

File: ntoskrnl/kd/kdmain.c [hunk 1]
```diff
@@ -153,11 +153,39 @@ KdpEnterDebuggerException(IN PKTRAP_FRAME TrapFrame,
...
+                }
+                _SEH2_EXCEPT(EXCEPTION_EXECUTE_HANDLER)
+                {
+                    UnsafeSymbolsInfo = NULL;
+                }
+                _SEH2_END;
+            }
+            else
+            {
+                SymbolsInfo = *UnsafeSymbolsInfo;
```
Comment (learn-more) [inline ntoskrnl/kd/kdmain.c]: Just for my education: Why does this copy not use RtlCopyMemory, but inside the seh block the copy is done with RtlCopyMemory?

Comment (ThFabba) [reply ntoskrnl/kd/kdmain.c]: The RtlCopyMemory was a spontaneous change to make sure the assignment doesn't get optimized out. It's probably not necessary, but in either case it certainly should be consistent between the two cases.

File: ntoskrnl/kd/kdmain.c [hunk 2]
```diff
@@ -153,11 +153,39 @@ KdpEnterDebuggerException(IN PKTRAP_FRAME TrapFrame,
...
+            }
+
+            if (UnsafeSymbolsInfo != NULL)
+            {
+                /* Load symbols. Currently implemented only for KDBG! */
+                if(KdbpSymFindModule(SymbolsInfo.BaseOfDll, NULL, -1, &LdrEntry))
```
Comment (HeisSpiter) [inline ntoskrnl/kd/kdmain.c]: Coding style, Thomas! :-(
OK, it's a copy paste, but still! :-p

File: ntoskrnl/kd/kdio.c [hunk 3]
```diff
@@ -567,14 +567,38 @@ KdpScreenInit(PKD_DISPATCH_TABLE DispatchTable,
...
     if (!KdpDebugMode.Value) return 0;
 
+    Length = min(Length, 512);
+
+    if (ExGetPreviousMode() != KernelMode)
+    {
+        _SEH2_TRY
+        {
+            ProbeForRead(UnsafeString, Length, 1);
+            String = _alloca(Length + 1);
```
Comment (learn-more) [inline ntoskrnl/kd/kdio.c]: Does _alloca play nice with seh?

Comment (ThFabba) [reply ntoskrnl/kd/kdio.c]: Hmm not sure. Well enough to pass the Wine test at least *shrug*

Comment (learn-more) [reply ntoskrnl/kd/kdio.c]: ship it!

**Pull request merged**